build.assets/update-api-module-path/main.go

```diff
@@ -0,0 +1,273 @@
+	if err := updateModFile(modulePath, oldModPath, newModPath); err != nil {
+		return trace.Wrap(err, "failed to update mod file for module")
+	}
+
+	return nil
+}
+
+// updateGoImports updates instances of the oldModPath with the newModPath in the given package.
+func updateGoImports(p *packages.Package, oldModPath, newModPath string) error {
+	for _, syn := range p.Syntax {
+		var rewritten bool
+		for _, i := range syn.Imports {
+			imp := strings.Replace(i.Path.Value, "\"", "", 2)
+			if strings.HasPrefix(imp, oldModPath) && !strings.HasPrefix(imp, newModPath) {
+				newImp := strings.Replace(imp, oldModPath, newModPath, 1)
+				if astutil.RewriteImport(p.Fset, syn, imp, newImp) {
+					rewritten = true
+				}
+			}
+		}
+		if !rewritten {
+			continue
+		}
+
+		goFileName := p.Fset.File(syn.Pos()).Name()
+		f, err := os.Create(goFileName)
+		if err != nil {
+			return errors.Wrapf(err, "could not create go file %v", goFileName)
+		}
+		defer f.Close()
+
+		if err = format.Node(f, p.Fset, syn); err != nil {
+			return errors.Wrapf(err, "could not rewrite go file %v", goFileName)
+		}
+	}
+
+	return nil
+}
+
+// updateModFile updates instances of oldPath to newPath in a go.mod file.
+// The modFile is updated in place by updating the syntax fields directly.
+func updateModFile(dir, oldPath, newPath string) error {
+	modFile, err := getModFile(dir)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	// Update mod name if needed
+	if modFile.Module.Syntax.Token[1] == oldPath {
+		modFile.Module.Syntax.Token[1] = newPath
+	}
+	// Update require statements in place if needed
+	for _, r := range modFile.Require {
+		if r.Mod.Path == oldPath {
+			pathI, versionI := 0, 1
+			if r.Syntax.Token[0] == "require" {
+				pathI, versionI = 1, 2
+			}
+			r.Syntax.Token[pathI], r.Syntax.Token[versionI] = newPath, currentVersionString()
+		}
+	}
+	// Update replace statements in place if needed
+	for _, r := range modFile.Replace {
+		if r.Old.Path == oldPath {
+			pathI := 0
+			if r.Syntax.Token[0] == "replace" {
+				pathI = 1
+			}
+			r.Syntax.Token[pathI] = newPath
+		}
+	}
+	// Format and save mod file
+	bts, err := modFile.Format()
+	if err != nil {
+		return errors.Wrap(err, "could not format go.mod file with new import path")
+	}
+	if err = ioutil.WriteFile(filepath.Join(dir, "go.mod"), bts, 0660); err != nil {
+		return errors.Wrap(err, "could not rewrite go.mod file")
+	}
+	return nil
+}
+
+// updateFiles updates instances of the oldModPath with the newModPath
+// in files with the given fileExtension in the given directory.
+func updateFiles(dir, fileExtension, oldModPath, newModPath string) error {
+	return filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if strings.HasSuffix(d.Name(), fileExtension) {
+			data, err := ioutil.ReadFile(path)
+			if err != nil {
+				return err
+			}
+			fileString := strings.ReplaceAll(string(data), oldModPath, newModPath)
```

Comment:
No need to go from `[]byte` to `string` and back again. Just use `bytes.ReplaceAll` instead.